src/components/organisms/NavBarSchedule/NavBarSchedule.tsx

```diff
@@ -97,21 +110,88 @@ export const NavBarSchedule: React.FC<NavBarScheduleProps> = ({
     toggle: togglePersonalisedSchedule,
   } = useShowHide(false);
 
+  const liveAndFutureEvents = useMemo(
+    () =>
+      relatedVenueEvents.filter(
+        (event) =>
+          secondsToMilliseconds(
+            event.start_utc_seconds + minutesToSeconds(event.duration_minutes)
+          ) > startOfToday().getTime()
+      ),
+    [relatedVenueEvents]
+  );
+
+  const liveEventsMinimalStartValue = Math.min(
+    ...liveAndFutureEvents.map((event) => event.start_utc_seconds)
+  );
+
+  const firstLiveEvent = liveAndFutureEvents.find(
+    (event) => event.start_utc_seconds === liveEventsMinimalStartValue
+  );
+
+  const minDate = useMemo(
+    () =>
+      firstLiveEvent
+        ? isEventLiveOrFuture(firstLiveEvent)
+          ? new Date(liveEventsMinimalStartValue).getTime()
+          : millisecondsToSeconds(todaysDateTime)
+        : minRangeValue,
+    [firstLiveEvent, liveEventsMinimalStartValue]
+  );
+
+  const firstRangeDayTimeValue = isDateLessOrEqualsToday({
+    valueSource: secondsToMilliseconds(minDate),
+    valueTarget: millisecondsToSeconds(startOfToday().getTime()),
+  })
+    ? millisecondsToSeconds(todaysDate.getTime())
+    : millisecondsToSeconds(new Date(secondsToMilliseconds(minDate)).getTime());
+
+  const maxDate = useMemo(
+    () =>
+      Math.max(
+        ...liveAndFutureEvents.map(
+          (event) =>
+            event.start_utc_seconds + minutesToSeconds(event.duration_minutes)
+        ),
+        firstRangeDayTimeValue + 1
```

Comment:
why +1?

Reply:
+ 1 is needed to form a `daysInBetween` timeline and mitigate possible range error

Reply:
could you add a comment, describing that?

Reply:
yep, added comment in b41ef61